process_report/invoices/NERC_total_invoice.py

```diff
@@ -6,6 +6,12 @@
 
 @dataclass
 class NERCTotalInvoice(invoice.Invoice):
+    """
+    Dependancies:
+    - ValidateBillablePIsProcessor
+    - NewPICreditProcessor
+    """
```

Comment:
What are these dependencies? Are we documenting them here (and elsewhere in this file) because they are otherwise non-obvious?

Reply:
Many of the invoice subclasses expect certain columns to exist in the dataframe they are given, and some of those columns are initialized and populated by certain processors. `ValidateBillablePIsProcessor` populates the `IS_BILLABLE_FIELD` and `MISSING_PI_FIELD`, while `NewPICreditProcessor` populates many columns, such as `BALANCE_FIELD`. These columns are used by many invoice classes, including `NERCTotalInvoice`, during their filtering or exporting steps. As such, I believed that there are dependencies between invoices and processors, since many invoices need certain processors to run before them to work properly.

I thought that I should highlight these dependencies with these docstrings. I will follow your advice if you believe something better should be done.

Reply:
I think it's fine to document this in the class docstring, but I think you should be a little bit more verbose. E.g., `This class requires the following columns to be defined in the input dataframe:`. This makes it clear that you're not referring to Python modules or classes or something else that could be considered a "dependency". If the required fields are only provided by a specific processor, it's probably better to say something like `This class operates on data that has been processed by NewPICreditProcess`.

Also note that the spelling is "dependencies".